src/useVegaEmbed.ts

```diff
@@ -0,0 +1,41 @@
+import vegaEmbed, { VisualizationSpec, Result, EmbedOptions } from 'vega-embed'
+import { computed, Ref, ref } from 'vue-demi'
+
+import { ModifyFunction } from './utils'
+
+/**
+ * A low level hook that simply wraps the vega embed.
+ *
+ * Properties are not dynamically update.
+ */
+export function useVegaEmbed(
+  el: Ref<HTMLElement | null | undefined>,
+  spec: Ref<VisualizationSpec>,
+  options?: Ref<EmbedOptions>
+) {
+  const result = ref<Result | null>(null)
+  const view = computed(() => result.value?.view ?? null)
+
+  const clear: () => void = () =>
+    result.value != null && result.value.finalize()
```

Comment:
Great that you do cleanup. Note that embed has its own finalize and you should use it since it cleans up some additional listeners. See https://github.com/vega/vega-embed#embed

Reply:
Isn't that the `finalize()` function I am calling, since
```ts
result.value = await vegaEmbed(el.value, spec.value, options?.value)
```

Reply:
You're right. I read `view.finalize`.